expression/builtin_time_vec.go

```diff
@@ -505,11 +505,49 @@ func (b *builtinSubDatetimeAndStringSig) vecEvalTime(input *chunk.Chunk, result
 }
 
 func (b *builtinDurationDurationTimeDiffSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDurationDurationTimeDiffSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalDuration(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ResizeGoDuration(n, false)
+	r64s := result.GoDurations()
+	d64s := buf.GoDurations()
+	d64s1 := buf1.GoDurations()
+	result.MergeNulls(buf, buf1)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		lhs := types.Duration{Duration: d64s[i]}
+		rhs := types.Duration{Duration: d64s1[i]}
```

Comment:
we can define `lhs` and `rhs` outside the `for` loop, which saves two allocation for each loop iteration. It would improve the performance IMHO.

Reply:
I will fix it.

Reply:
I will fix it.